websites/C/ComicK/presence.ts

```diff
@@ -1,128 +1,107 @@
+					0,
+					img.alt.indexOf("chapter")
+				)}`;
+				presenceData.state = img.alt.substring(
+					img.alt.indexOf("chapter"),
+					img.alt.indexOf(",")
+				);
+				presenceData.largeImageKey = document.querySelector<HTMLMetaElement>(
+					"meta[property='og:image']"
+				).content;
+				presenceData.buttons = [
+					{
+						label: "Read Chapter",
+						url: href,
+					},
+					{
+						label: "Read Description",
+						url: href.split(/(.+)[\\/]/)[1],
+					},
+				];
 			}
+			break;
 		}
-	} else if (pathname.startsWith("/group")) {
-		presenceData.details = "Looking at group";
-		presenceData.state =
-			document.querySelector<HTMLHeadingElement>("h1").textContent;
-	} else if (pathname.startsWith("/search")) {
-		presenceData.details = "Searching";
-		for (const t of document.querySelectorAll("h1.mb-3 > div")) {
-			if (t.textContent) {
-				presenceData.state = t.textContent;
-				break;
-			}
-		}
-	} else if (pathname.startsWith("/user") && pathname !== "/user") {
-		presenceData.details = `Viewing ${
-			document.querySelector<HTMLHeadingElement>("h1").textContent
-		}`;
-		if (image) {
-			presenceData.largeImageKey = document
-				.querySelector<HTMLImageElement>("#__next div > div > img")
-				.src.replace("size=200", "size=640");
+		case "group":
+			presenceData.details = "Looking at group";
+			presenceData.state =
+				document.querySelector<HTMLHeadingElement>("h1").textContent;
+			break;
+		case "search": {
+			presenceData.details = "Searching";
+			const research = document.querySelector("div > h1")?.textContent;
+			if (research) presenceData.state = research;
+			break;
 		}
-	} else {
-		switch (pathname) {
-			case "/ranking": {
-				presenceData.details = "Looking at rankings";
-				break;
-			}
-			case "/comment_list": {
-				presenceData.details = "Looking at comment list";
-				break;
+		case "user":
+			presenceData.details = "Viewing their profile";
+			if (arrPath.length > 2) {
+				presenceData.details = `Viewing ${
+					document.querySelector<HTMLHeadingElement>("h1").textContent
+				}`;
+				presenceData.largeImageKey = document
+					.querySelector<HTMLImageElement>("#__next div > div > img")
+					.src.replace("size=200", "size=640");
 			}
-			case "/settings": {
-				presenceData.details = "Settings";
-				break;
-			}
-			case "/languages": {
-				presenceData.details = "Languages";
-				break;
-			}
-			case "/privacy": {
-				presenceData.details = "Privacy POlicy";
-				break;
-			}
-			case "/install_app": {
-				presenceData.details = "ComicK App";
-				break;
-			}
-			case "/user": {
-				presenceData.details = "Viewing their profile";
-				break;
-			}
-		}
+			break;
+		default:
+			if (Object.keys(staticPages).includes(arrPath[1]))
+				presenceData = { ...presenceData, ...staticPages[arrPath[1]] };
 	}
+
+	if (!image && presenceData.largeImageKey !== Assets.Logo)
+		presenceData.largeImageKey = Assets.Logo;
 	if (!buttons) delete presenceData.buttons;
```

Comment:
Same here, check if there are button (optional)

Reply:
I'll add it when I'll add strings